context_header.go

```diff
@@ -114,3 +114,19 @@ func WrapWithHeaders(ctx context.Context, headers map[string]string) ContextWith
 	newCtx := context.WithValue(ctx, contextKeyHeaders, h)
 	return headerCtx{Context: newCtx}
 }
+
+// WithoutHeaders hides any TChannel headers from the given context.
+func WithoutHeaders(ctx context.Context) context.Context {
```

Comment:
The real thing we're trying to do here is make sure `ctx` is not a `ContextWithHeaders`, so we want to wrap regardless